docs/examples/openapi/plugins/custom_plugin.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import Any
+
+from litestar.connection import Request
+from litestar.openapi.plugins import OpenAPIRenderPlugin
+
+
+class RapidocRenderPlugin(OpenAPIRenderPlugin):
+    """Render an OpenAPI schema using Rapidoc."""
+
+    def __init__(
+        self,
+        *,
+        version: str = "9.3.4",
+        js_url: str | None = None,
+        path: str | Sequence[str] = "/rapidoc",
+        **kwargs: Any,
+    ) -> None:
+        """Initialize the OpenAPI UI render plugin.
+
+        Args:
+            version: Rapidoc version to download from the CDN. If js_url is provided, this is ignored.
+            js_url: Download url for the RapiDoc JS bundle. If not provided, the version will be used to construct the
+                url.
+            path: Path to serve the OpenAPI UI at.
+            **kwargs: Additional arguments to pass to the base class.
+        """
+        self.js_url = js_url or f"https://unpkg.com/rapidoc@{version}/dist/rapidoc-min.js"
+        super().__init__(path=path, **kwargs)
+
+    def render(self, request: Request, openapi_schema: dict[str, Any]) -> bytes:
+        """Render an HTML page for Rapidoc.
+
+        Notes:
+            - override this method to customize the template.
+
+        Args:
+            request: The request.
+            openapi_schema: The OpenAPI schema as a dictionary.
+
+        Returns:
+            A rendered html string.
+        """
+
+        head = f"""
+          <head>
+            <title>{openapi_schema["info"]["title"]}</title>
+            {self.favicon}
+            <meta charset="utf-8"/>
+            <meta name="viewport" content="width=device-width, initial-scale=1">
+            <script src="{self.js_url}" crossorigin></script>
+            {self.style}
+          </head>
+        """
+
+        body = """
+          <body>
+            <rapi-doc spec-url="openapi.json" />
```

Comment:
sorry if this is dumb, but is this an accurate path considering https://github.com/litestar-org/litestar/pull/3196

Reply:
Yeh not sure - thanks for pointing out. @guacs can you help me out with this one please - all I've done is transfer the logic we had on the controller routes into the individual plugins so a good chance I need to do something differently.

Reply:
related: https://github.com/litestar-org/litestar/pull/2893#discussion_r1437898435

Reply:
We also have same in the scalar plugin:

```py
        body = f"""
                <noscript>
                    Scalar requires Javascript to function. Please enable it to browse the documentation.
                </noscript>
                <script
                  id="api-reference"
                  data-url="openapi.json">
                </script>
                <script src="{self.js_url}" crossorigin></script>
                ""

Reply:
I've found it works if you request `/schema/` (with trailing `/`) but not `/schema` - I suppose I have to work out how to implement something similar to #3196.

Reply:
OK sorry about the chatter on this one. I've cherry picked #3196 into this branch and then pretty much implemented the same thing in the context of the openapi router/plugins. @guacs would you mind taking a look at https://github.com/litestar-org/litestar/pull/2893/commits/8f1344701b588785ad0639392385116d0cf31852